plugin/external/pluginRoot.go

```diff
@@ -0,0 +1,75 @@
+// Package external defines types to represent the structure of plugin implemented
```

Comment:
Now that I think about it, I kind of like the old structure better, where external plugin stuff was in the `plugin` package (even though I accidentally ended up using that structure due to a dependency cycle that this PR solves). The reason is b/c an external plugin's implementation is internal to Wash and is not something that people should touch. In fact, the only relevant bit that should be exported (and documented) is `ExternalPluginSpec`.

Right now, `NewExternalPluginRoot` needs to be exported, but I think we can clean that up by adding an extra method to the plugin registry (`RegisterExternalPlugin(ExternalPluginSpec)`). That can be a separate PR though.

Reply:
I'll explore making more of them private types.

Reply:
Removed.